File: packages/javascript/lib/verify/isEmpty.ts
export function isEmpty(data?: any): boolean {
	if (!data) {
		return true;
	}

	switch (typeof data) {
		case 'string':
			return data.length < 1;
		case 'object':
			if (Array.isArray(data)) {
				return data.length < 1;
			}
			return Object.keys(data).length < 1;
		default:
			return false;
	}
}

File: packages/python/qsu/verify/isEmpty.py
def isEmpty(data=None) -> bool:
	if not data:
		return True

	if isinstance(data, str):
		return len(data) < 1
	if isinstance(data, (list, tuple)):
		return len(data) < 1
	if isinstance(data, dict):
		return len(data.keys()) < 1

	return False
